application-templates/starter/src/components/channel-details/channel-details-form.js

```diff
@@ -0,0 +1,104 @@
+import PropTypes from 'prop-types';
+import { useFormik } from 'formik';
+import { useIntl, FormattedMessage } from 'react-intl';
+import { useApplicationContext } from '@commercetools-frontend/application-shell-connectors';
+import LocalizedTextField from '@commercetools-uikit/localized-text-field';
+import TextField from '@commercetools-uikit/text-field';
+import Spacings from '@commercetools-uikit/spacings';
+import { useIsAuthorized } from '@commercetools-frontend/permissions';
+import { ErrorMessage } from '@commercetools-uikit/messages';
+import SelectField from '@commercetools-uikit/select-field';
+import { PERMISSIONS } from '../../constants';
+import { CHANNEL_ROLES } from './constants';
+import validate from './validate';
+import messages from './messages';
+
+const getRoleOptions = (intl) =>
+  Object.keys(CHANNEL_ROLES).map((key) => ({
+    label: intl.formatMessage(messages[key]),
+    value: CHANNEL_ROLES[key],
+  }));
+
+const ChannelDetailsForm = (props) => {
+  const intl = useIntl();
+  const { dataLocale } = useApplicationContext((context) => ({
+    dataLocale: context.dataLocale,
+  }));
+  const formik = useFormik({
+    initialValues: props.initialValues,
+    onSubmit: props.onSubmit,
+    validate,
+    enableReinitialize: true,
+  });
+  const canManage = useIsAuthorized({
+    demandedPermissions: [PERMISSIONS.Manage],
+  });
+
+  const formElements = (
+    <Spacings.Stack scale="l">
+      <Spacings.Stack scale="s">
+        <TextField
+          name="key"
+          title={intl.formatMessage(messages.channelKeyLabel)}
+          value={formik.values.key}
+          errors={formik.errors.key}
+          touched={formik.touched.key}
+          onChange={formik.handleChange}
+          onBlur={formik.handleBlur}
+          isReadOnly={!canManage}
+          isRequired={canManage}
```

Comment:
I would keep it as-is, no matter the user permissions

```suggestion
          isRequired
```